server/backend/database/mongo/client.go

```diff
@@ -643,6 +643,35 @@ func (c *Client) UpdateClientInfoAfterPushPull(
 	return nil
 }
 
+// DeleteUserInfoByName deletes a user by name.
+func (c *Client) DeleteUserInfoByName(ctx context.Context, username string) error {
+	deleteResult, err := c.collection(ColUsers).DeleteOne(ctx, bson.M{
+		"username": username,
+	})
+	if err != nil {
+		return err
+	}
+	if deleteResult.DeletedCount == 0 {
+		return fmt.Errorf("nothing has been deleted")
```

Comment:
How about enhancing error message like `fmt.Errorf("no user found with username %s", username)`?
We want to keep consistency for all the error messages in this file. Refer to other error messages for information.

Reply:
oh that's good idea. I miss that thank you! :)